edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/AvaliabilityMetrics.cs

```diff
@@ -41,6 +47,8 @@ public AvailabilityMetrics(IMetricsProvider metricsProvider, ISystemTime time)
             this.time = time;
```

Comment:
move lines for this.running and this.expectedRunning assignment after this.edgeAgent assignment (line 49); create an instance after validation.